content/2. Ondersteunende-informatie/3. Processen/6. Beheerproces/4. Beschikbaarstellen software/OTAP ontwikkeling, test, acceptatie en productie.md

```diff
@@ -0,0 +1,22 @@
+
+## Waarom gebruik je OTAP?
+Werken volgens de OTAP-straat is heel belangrijk als je software ontwikkelt. Je wilt niet dat de eindgebruiker last heeft van tests of een onvolledige versie van de software te zien krijgt, inclusief mogelijke bugs. Door de OTAP-methode hebben Developers de ruimte en tijd om hun werk te doen en functies uitgebreid te testen voordat de software in productie gaat. Ook kan de opdrachtgever hierdoor feedback geven op de software voordat deze in gebruik wordt genomen.
+
+## Hoe gebruik je OTAP?
+Door vier verschillende omgevingen in te richten en die te markeren voor ontwikkeling, test, acceptatie of productie, is je OTAP-straat ingericht. De verschillende omgevingen werken met hun eigen versie van data, maar bij grote systemen ook op hun eigen infrastructuur! In de praktijk worden soms de ontwikkel en testomgeving gecombineerd vanwege de overhead die de testomgeving geeft en de ontwikkelaars het ook goed getest willen hebben (met dezelfde software als die waarmee getest wordt).
+
+Als ontwikkelaar moet je er zelf voor zorgen dat de software wordt overgezet naar de juiste plek. Dan kan geautomatiseerd door tooling te gebruiken of de Visual Studio omgeving hierop aan te passen (b.v. deployment naar Azure kan gemakkelijk vanuit de IDE) . Om dit mogelijk te maken heb je een buildstraat nodig waarmee je je eigen onderdelen kunt bouwen en testen die vervolgens kunt samenvoegen met componenten van je teamgenoten (of andere ontwikkelaars).
+Een goed model hiervoor zie je in het onderstaande plaatje:
+![[Buildstraat.png]]
+
+De laatste stap in dit proces is deploy. Als alle testen slagen kan de software worden doorgezet naar de testomgeving. Als de software door de testomgeving komt kan het worden doorgezet naar de acceptatieomgeving etc.
+C# werk middels xcopy deployment principe. Met andere woorden de software kan eenvoudig vanuit de bin\release map worden gekopieerd naar de juiste plek.
```

Comment:
Misschien een idee om een verwijzing naar de (externe) bron te maken. In de andere content is de Microsoft documentatie vaker gebruikt